activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -33,6 +33,10 @@ def deserialize(value)
       end
 
       def serialize(value)
+        cast(value)
+      end
+
+      def cast(value)
```

Comment:
Implemented `#cast` to make the optimization work, but this feels off. I need to look into all this more calmly...